src/main/webapp/scripts/sessioncache.js

```diff
@@ -0,0 +1,117 @@
+/** 
+ * SessionCache bridges the gap between the client and server.
+ * Allows client to indirectly stay in contact with the server,
+ * encapsulating many of the contact points away from the client.
+ * 
+ * SessionCache looks for information about the current session and
+ * updates the server with information about each client in return (like
+ * date last contacted). Also includes some additional functionality, 
+ * like contacting the server to update the controller of the session.
+ */
+class SessionCache {
+  /**
+   * Initalizes a SessionCache object.
+   * @param {Object} urlParams Represents the URLSearchParams of the
+   *    session the client is in, holds information such as the
+   *    session ID and the screen name of the current user.
+   */
+  constructor(urlParams) {
+    /** 
+     * Poll responsible for contacting the server for information about
+     * the current session.
+     * @private {Object} 
+     */
+    this.sessionInformationPoll_ = null;
+
+    /** 
+     * Poll responsible for updating the server of information about Date
+     * last polled. 
+     * @private {Object} 
+     */
+    this.trackForInactivityPoll_ = null;
```

Comment:
Why? The already server knows the last time it was polled, right? Since it must've handled the request. Why does it need to be updated? Also, you dont need more than one poll (or poller). A poll for session information should be enough